src/App.js

```diff
@@ -1,26 +1,205 @@
+      });
+  }
+
+  componentDidMount() {
+    this.getElectionEvents();
+  }
+
+  render() {
+    return (
+      <ul>
+        {this.state.election_events.map(function(item) {
+          return <li key={item.id}>{item.name}</li>;
+        })}
+      </ul>
+    );
+  }
+}
+
+class Elections extends React.Component {
+  state = {
+    elections: []
+  };
+
+  handleError(res, message) {
+    console.log("ERROR (" + res.status + "): " + message);
+  }
+
+  getElections() {
+    fetch("https://sleepy-retreat-45150.herokuapp.com/api/elections")
+      .then(response => {
+        if (!response.ok) {
+          this.handleError(response, "Failed to get elections");
+        }
+        return response.json();
+      })
+      .then(data => {
+        this.setState({ elections: data.elections });
+      });
+  }
+
+  componentDidMount() {
+    this.getElections();
+  }
+
+  render() {
+    return (
+      <ul>
+        {this.state.elections.map(function(item) {
+          return (
+            <li key={item.id}>
+              {item.election_event} {item.election_type}
+            </li>
+          );
+        })}
+      </ul>
+    );
+  }
+}
+
+class BarChart extends React.Component {
+  componentDidMount() {
+    this.svg = ReactDOM.findDOMNode(this);
+    this.drawChart();
+  }
+
+  componentDidUpdate() {
+    const { width, height } = this.props;
+    if (width !== 0 && height !== 0) {
+      this.drawChart();
+    }
+  }
+
+  drawChart() {
+    const { data, height } = this.props;
+
+    this.svg.innerHTML = "";
+
+    const svg = d3.select(this.svg);
+
+    svg
+      .selectAll("rect")
+      .data(data)
+      .enter()
+      .append("rect")
+      .attr("x", (d, i) => i * 90) // space between columns
+      .attr("y", (d, i) => height - 10 * d)
+      .attr("width", 65)
+      .attr("height", (d, i) => d * 10)
+      .attr("fill", "gray");
+  }
+
+  render() {
+    const { width, height } = this.props;
+    return <svg width={width} height={height} />;
+  }
+}
+
+class App extends React.Component {
+  state = {
+    data: [12, 5, 6, 6, 9, 10],
+    map_data: [12, 5, 6, 6, 9, 10],
+    width: 700,
+    height: 200,
+    map_width: 800,
+    map_height: 450
+  };
+
+  handleClick = (e, countyNumber) => {
+    console.log("Click happened :", countyNumber);
+  };
+
+  handleMouseOver = (e, text) => {
+    console.log("MouseOver happened :", text);
+  };
+
+  handleMouseOut = e => {
+    console.log("MouseOut happened");
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <header className="App-header">
+          <p>Election Results for 2019</p>
+          <Norway
+            id="simple"
+            width={2 * this.state.map_width}
+            height={2 * this.state.map_height}
+            onClick={this.handleClick}
+            onMouseOver={this.handleMouseOver}
+            onMouseOut={this.handleMouseOut}
+          />
+          <BarChart data={this.state.data} width={this.state.width} height={this.state.height} />
```

Comment:
I'm not sure what feedback is best to give here as I don't know the end desired UI, but likely it'd make sense for the `this.state.data` to be owned by the BarChart; If the goal is to have the map control the barchart, then you could do something like having a "countyNumber" in state, and then have <BarChart> accept a "countyNumber" prop, which then triggers the fetch for it's data.

For the various widths, I'm not sure what the intent here is, but if it's to make it responsive, you'll want to debounce or throttle changes to those props, or perhaps look at actually using percentage-based units rather than fixed-size units (px, em, etc)